opentelemetry-appender-log/src/lib.rs

```diff
@@ -70,13 +74,569 @@ const fn severity_of_level(level: Level) -> Severity {
     }
 }
 
-#[cfg(all(test, feature = "testing", feature = "logs"))]
+fn log_attributes(kvs: impl log::kv::Source) -> Vec<(Key, AnyValue)> {
+    struct AttributeVisitor(Vec<(Key, AnyValue)>);
+
+    impl<'kvs> log::kv::VisitSource<'kvs> for AttributeVisitor {
+        fn visit_pair(
+            &mut self,
+            key: log::kv::Key<'kvs>,
+            value: log::kv::Value<'kvs>,
+        ) -> Result<(), log::kv::Error> {
+            let key = Key::from(String::from(key.as_str()));
+
+            if let Some(value) = any_value::serialize(value) {
+                self.0.push((key, value));
+            }
+
+            Ok(())
+        }
+    }
+
+    let mut visitor = AttributeVisitor(Vec::new());
+
+    let _ = kvs.visit(&mut visitor);
+
+    visitor.0
+}
+
+#[cfg(not(feature = "with_serde"))]
+mod any_value {
+    use opentelemetry::{logs::AnyValue, StringValue};
+
+    pub(crate) fn serialize(value: log::kv::Value) -> Option<AnyValue> {
+        struct ValueVisitor(Option<AnyValue>);
+
+        impl<'kvs> log::kv::VisitValue<'kvs> for ValueVisitor {
+            fn visit_any(&mut self, value: log::kv::Value) -> Result<(), log::kv::Error> {
+                self.0 = Some(AnyValue::String(StringValue::from(value.to_string())));
+
+                Ok(())
+            }
+
+            fn visit_bool(&mut self, value: bool) -> Result<(), log::kv::Error> {
+                self.0 = Some(AnyValue::Boolean(value));
+
+                Ok(())
+            }
+
+            fn visit_str(&mut self, value: &str) -> Result<(), log::kv::Error> {
+                self.0 = Some(AnyValue::String(StringValue::from(value.to_owned())));
```

Comment:
Same as key, if log's value provides the way to check if the string is borrowed or owned ?

Reply:
This is another case where it doesn’t, but we’ve left room for it.

Reply:
Looks like it does [support](https://github.com/rust-lang/log/blob/7bdb8aac88c848df2668149d67a6acc971e1a8e0/src/kv/value.rs#L992), but should be fine to support it as separate PR.

Reply:
Ah yes, that method lets us attempt to pull a string for a specific lifetime `'v` from a `Value<'v>`, but for `StringValue` we'd need that lifetime to be `'static`. It's _almost_ what we'd need here.